File: apple/Shared/Models/Insight.swift
import Foundation

/// An AI-generated insight: a summary, theme, or suggestion.
struct Insight: Codable, Identifiable {
    let id: UUID
    let fragmentId: UUID?
    let insightType: InsightType
    let content: String
    let metadata: [String: String]
    let createdAt: Date

    enum InsightType: String, Codable {
        case summary
        case theme
        case suggestion
        case relatedTopic
    }
}

File: backend/src/models/insight.ts
import { z } from "zod";

export const InsightTypeEnum = z.enum([
  "summary",
  "theme",
  "suggestion",
  "relatedTopic",
]);
export type InsightType = z.infer<typeof InsightTypeEnum>;

export interface Insight {
  id: string;
  userId: string;
  fragmentId: string | null;
  insightType: InsightType;
  content: string;
  metadata: Record<string, unknown>;
  createdAt: Date;
}
